sql/core/pom.xml

```diff
@@ -151,6 +151,10 @@
       <artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>${hive.group}</groupId>
+      <artifactId>hive-exec</artifactId>
+    </dependency>
```

Comment:
We should remove this in this PR.

Reply:
Main reason of `HiveResult` presents in `sql/core` is because it is used in `SQLQueryTestSuite`: https://github.com/apache/spark/blob/2a30deb85ae4e42c5cbc936383dd5c3970f4a74f/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L290 If we get rid off using it in the tests, we can move it to `hive`.

Reply:
done

Reply:
is this really done? I thought this should have been moved to the hive module...

Reply:
It is hard to move it there because `hiveResultString ` is used in `SQLQueryTestSuite`: https://github.com/apache/spark/blob/2a30deb85ae4e42c5cbc936383dd5c3970f4a74f/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L290 . If you move it to `hive`, `sql/test` will depend on `hive` but `hive` depends on `sql` which makes `mvn`/`sbt` mad.

Reply:
I removed `hive-exec` dependency here. So I think it should be ok don't move `HiveResult` to `hive` module.